app.js

```diff
@@ -231,19 +240,25 @@ app.post('/interactions', verifyKeyMiddleware(process.env.PUBLIC_KEY), async (re
             });
         }
         
-        else if (args[0] === 'track') {
+        else if (args[0].slice(0, 5) === 'track') {
+            const targs = args[0].split('_');
+            console.log(targs);
+            console.log(args);
+            const groupUid = args[1].split('-').map((e, i) => convertBase62ToNumber(e, GROUP_UID_LENS[i], 16)).join('-');
             let command;
             let track_info;
-            if (args[1] === 'totd') { 
-                command = `Track of the Day - ${args[4]}`;
-                track_info = await cachingTOTDProvider.getData().catch(err => embeddedErrorMessage(endpoint, err));
+            if (targs[1] === 'totd') { 
+                command = `Track of the Day - ${args[3]}`;
+                if (Number(convertBase62ToNumber(targs[2])) > Math.floor(Date.now() / 1000)) 
+                    track_info = await cachingTOTDProvider.getData().catch(err => embeddedErrorMessage(endpoint, err));
+                else track_info = await trackmaniaFacade.getTrackInfo(command, args[2], groupUid).catch(err => embeddedErrorMessage(endpoint, err));
```

Comment:
Once the data has expired and we fetch the data from the Trackmania API, this won't cache it, right? Also, this timestamp check logic is a little duplicated from the caching data provider ([code ptr](https://github.com/Khujou/trackmania-bot/blob/aae334a8c299402b2fc881f892d4d35362d5e229/app.js#L49)).

Could we create a `FileBasedCachingDataProvider` that takes the `mapUid` as a parameter to `getData()` so we can de-dupe the timestamp logic and leverage the existing cache logic? This data provider can just slowly build up a cache of every track we've seen so far and we can let it handle fetching new tracks as needed